plugins/uefi-dbx/fu-uefi-dbx-common.c

```diff
@@ -26,3 +26,63 @@ fu_uefi_dbx_get_dbxupdate (GError **error)
 		return NULL;
 	return g_strdup (g_ptr_array_index (files, 0));
 }
+
+gchar *
+fu_uefi_dbx_get_authenticode_hash (const gchar *fn, GError **error)
+{
+	const gchar *argv[] = { "pesign", "-P", "-i", fn, "-h", NULL };
```

Comment:
Does this by chance also work with `sbsigntool`?  If not, it's going to be an upward battle to get `pesign` pulled in by default in distros that don't use it.

Reply:
I didn't find any way of making the various `sbsigntool` command lines give me the authenticode hash of the EFI binary. If you know a way offhand that would be great, otherwise I can tap the wisdom of twitter.

Reply:
I don't know off hand.  Entirely possible it doesn't support it too!

One more thought - is it a lot of code to lift from pesign?  Maybe just borrow it directly into here?

Reply:
> is it a lot of code to lift from pesign

It looked a lot of code, but I can find out exactly how much tomorrow.

Reply:
`sbsigntools` doesn't expose the authenticode hash in any way. I wonder if it would be easier to contribute some tools to fetch it, they do implement their own PE/COFF parser library for it after all.

Reply:
Trammell Hudson implemented `--hash-only` for `sbsign`. I reckon this is what you wanted?

https://groups.io/g/sbsigntools/message/42

Reply:
That's really helpful, I've updated the patch to support `sbsign` as well now.